Eggstravaganza/Assets/Scripts/GameNetwork.cs

```diff
@@ -84,4 +102,53 @@ void EndRound()
             m_GameState.Value = GameState.EndRound;
         }
     }
+
+    void SpawnCountdown()
+    {
+        if (!IsOwner)
+            return;
+        
+        // Spawns an egg at a random time from 1 to maxEggSpawnTime seconds
+        if (m_TimerIsRunning)
+        {
+            if (m_TimeRemaining > 0)
+            {
+                m_TimeRemaining -= Time.deltaTime;
+            }
+            else
+            {
+                // Update timer
+                m_TimeRemaining = 0;
+                m_TimerIsRunning = false;
+
+                // Spawn chosen egg
+                RequestSpawnEggServerRpc();
+                m_EggSpawner.SpawnEgg(m_EggSpawnPosition.Value, m_EggToSpawnIndex.Value);
+        
+                // Choose next egg to spawn and its position
+                m_EggToSpawnIndex.Value = m_EggSpawner.ChooseEggToSpawn();
+                m_EggSpawnPosition.Value = m_EggSpawner.ChooseSpawnPosition();
+            }
+        }
+        else
+        {
+            m_TimeRemaining = Random.Range(0, maxEggSpawnTime);
+            m_TimerIsRunning = true;
+        }
+    }
+
+    [ServerRpc]
+    void RequestSpawnEggServerRpc()
+    {
+        SpawnEggClientRpc();
+    }
+
+    [ClientRpc]
+    void SpawnEggClientRpc()
+    {
+        if (!IsOwner)
+        {
+            m_EggSpawner.SpawnEgg(m_EggSpawnPosition.Value, m_EggToSpawnIndex.Value);
```

Comment:
just thinking aloud: i was wondering what the best solution is to track their positions considering physics? can we get away with the ClientNetworkTransform utility component on each egg? will we take a performance hit for tracking position + rotation? does rotation matter? is it just easiest if we hit like a max # eggs in the arena?

Reply:
tbh I don't quite undestand ClientNetworkTransform as well so I'm not sure of this, but won't we have to make every egg a network object to use ClientNetworkTransform? for now they aren't network objects based on the soccer video, but I'm open to changing it if needed! what do you think would that be better?

Reply:
ah yeah fair point; once they're picked up they're equivalent to the soccer balls i agree! that leaves what we do with them before picking them up. e.g. if an egg lands and rolls, should it be required to roll in the same direction over the network, such that it ends in the same position for everyone? or should they simply not be physics objects until picked up and thrown?

Reply:
> should it be required to roll in the same direction over the network, such that it ends in the same position for everyone?

this I would say yes, because since the egg has a weird shape if they roll to different positions for every client then there is a high chance different players would be able to pick it up on different clients which would make it messy imo. what do you think, do you have any other way in mind to make sure they land the same way?

Reply:
the only other thing i can think of is just letting them land extremely "solidly" so they don't move at all until being picked up, but i'm not sure it's even possible to flip a sort of NetworkObject switch during runtime? but i agree just syncing positions should be enough if performance becomes an issue

Reply:
discussed offline, we'll let the eggs be the way they are for now and check if any issues arise while merging with Amanda's code